### PR TITLE
MAP-1255 Add accessibility testing to integration tests by default

### PR DESCRIPTION
**Pull request by danbenton-mojdt**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -97,6 +97,7 @@
     "@ministryofjustice/frontend": "^2.1.3",
     "agentkeepalive": "^4.5.0",
     "applicationinsights": "^2.9.5",
+    "axe-core": "^4.7.2",
```
Comment (petergphillips) [inline package.json:100]: any reason for this to be a standard dependency instead of a dev dependency?

File: integration_tests/pages/page.ts [hunk 2]
```diff
@@ -1,19 +1,53 @@
...
     return new constructor()
   }
 
-  constructor(private readonly title: string) {
-    this.checkOnPage()
+  constructor(
+    private readonly title: string,
+    private readonly skipA11y = false,
+  ) {
+    if (title) {
```
Comment (petergphillips) [inline integration_tests/pages/page.ts:15]: why the check here?  Isn't `title` always going to be set?

File: package.json (hunk 1 above)
Comment (danbenton-mojdt) [reply package.json]: No, moved to dev dependencies

File: integration_tests/pages/page.ts (hunk 2 above)
Comment (danbenton-mojdt) [reply integration_tests/pages/page.ts]: Copied from other project without thinking about it much - removed!

File: integration_tests/pages/page.ts [hunk 3]
```diff
@@ -1,19 +1,51 @@
...
 
 export default abstract class Page {
   static verifyOnPage<T>(constructor: new () => T): T {
     return new constructor()
   }
 
-  constructor(private readonly title: string) {
+  constructor(
+    private readonly title: string,
+    private readonly skipA11y = false,
```
Comment (petergphillips) [inline integration_tests/pages/page.ts:13]: not a great fan of booleans in constructors - difficult to know what the `super('Sign in', true)` means without having a look at the implementation

File: integration_tests/pages/page.ts [hunk 4]
```diff
@@ -1,19 +1,51 @@
...
+  ) {
     this.checkOnPage()
   }
 
   checkOnPage(): void {
     cy.get('h1').contains(this.title)
+    if (!(Cypress.env('SKIP_AXE') || this.skipA11y)) {
```
Comment (petergphillips) [inline integration_tests/pages/page.ts:20]: we've got a `SKIP_AXE` and a `skipA11y`, wondering if it is better to have the same name for both i.e. `skipAxe` or maybe even `SKIP_ACCESSIBILITY_CHECKS` and `skipAccessibilityChecks`

**Review by petergphillips — APPROVED** [review]

File: integration_tests/pages/page.ts (hunk 3 above)
Comment (psoleckimoj) [reply integration_tests/pages/page.ts:13]: Would be easier as an object then at least you can have meaningful attribute names in there and then use a boolean.

**Pull request closed without merging**